src/viam/app/data_client.py

```diff
@@ -120,10 +120,15 @@ def __init__(self, channel: Channel, metadata: Mapping[str, str]):
         self._metadata = metadata
         self._data_client = DataServiceStub(channel)
         self._data_sync_client = DataSyncServiceStub(channel)
+        self._channel = channel
 
     _data_client: DataServiceStub
     _data_sync_client: DataSyncServiceStub
     _metadata: Mapping[str, str]
+    _channel: Channel
+
+    def close(self) -> None:
```

Comment:
That's a good point, we probably don't want users closing either just the app client or just the data client if either was accessed through a viam client. Some workarounds/ideas:
1. removing the app_client and data_client close methods (but keeping the viam_client close() method) - this would totally be fine. for scenarios where an appclient or dataclient is created directly (i.e., not through a viam client), we could just expect the user to manually app_client._channel.close() if they wanna avoid the annoying error message that always shows up if a channel isn't closed. Or I guess they could just rely on the GC, but that message still shows up.
2. Have two separate channels for app_client and data_client. it would allow for both to have their own close methods that wouldn't have any unexpected outcomes, but i'm not sure what other implications mainting two channels might have.
3. have an "open" method that undoes close(), I think this would be more complicated as we try to maintain both clients through the viam client.

the good news is that if a user creates an appclient directly, they don't really have to worry about this problem because they provide the channel in the constructor. So the issue is only if they use viamclient to create an appclient or dataclient but then use appclient.close or dataclient.close instead of viamclient.close

anyway with all that what do u think

(sorry for so many words lol)

Reply:
(resolved offline, removing the method)